xonsh/built_ins.py

```diff
@@ -720,6 +724,7 @@ def load_builtins(execer=None, config=None, login=False, ctx=None):
     builtins.__xonsh_execer__ = execer
     builtins.__xonsh_commands_cache__ = CommandsCache()
     builtins.__xonsh_all_jobs__ = {}
+    builtins.__xonsh_enabled_builtin_commands__ = set(all_builtin_commands.keys())
```

Comment:
I think so.  Most (?) other shells have some notion of built-in commands, which is somewhat distinct from coreutils.  That said, I think the vision is actually moving away from just implementing coreutils, toward better handling of built-in commands more generally.